mockserver/main.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"path"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+const (
+	HealthCheckMessage = "healthcheck"
+	SuccessMessage     = "success"
+)
+
+var (
+	CertFilePath = path.Join("certificates", "ssl", "certificate.crt")
+	KeyFilePath  = path.Join("certificates", "private.key")
+)
+
+type transactionHttpServer struct {
+	transactions uint32
+	startTime    time.Time
+}
+
+type TransactionPayload struct {
+	TransactionsPerMinute float64 `json:"GetNumberOfTransactionsPerMinute"`
+}
+
+func healthCheck(w http.ResponseWriter, _ *http.Request) {
+	if _, err := io.WriteString(w, HealthCheckMessage); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Printf("Unable to write response: %v", err)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (ts *transactionHttpServer) checkTransactionCount(w http.ResponseWriter, _ *http.Request) {
+	var message string
+	var t = atomic.LoadUint32(&ts.transactions)
+	if t > 0 {
+		message = SuccessMessage
+	}
+	log.Printf("\033[31m Time: %d | checkTransactionCount msg: %s | %d\033[0m \n", time.Now().Unix(), message, t)
+	if _, err := io.WriteString(w, message); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		io.WriteString(w, err.Error())
+		log.Printf("Unable to write response: %v", err)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (ts *transactionHttpServer) acknowledgeTransaction(w http.ResponseWriter, _ *http.Request) {
```

Comment:
nit: `acknowledgeTransaction` seems inaccurate. It's incrementing a counter. Consider something like `recordTransaction` or even `incrementTransactionCount`